### DISCUSSION TITLE
Lorawan & Lilygo T-beam V1.1 sx1262  "Failed, code 2745"

### DISCUSSION THREAD

**Discussion by icicmoi** [jgromes/RadioLib#1009] [General]
Hi all,
I've been trying for a long time to get a Tbeam "T22 V1.1" Lyligo module to work (with an sx1262 chip) ... but I have to face the fact that I'll never be a computer scientist but simply a curious but poor tinkerer ;)
I don't know how ... and am still unable to understand the information available in the arduino ide monitor (v 2.3.2).
So thank you very much for your help: I'm putting the code below (taken from your example) and also the debug info.

ps : at the end of the debug I note " failed, code 2745" ... but I haven't found out what it means (?)

---

[Debug_T-Beam_SX1262.txt](https://github.com/jgromes/RadioLib/files/14550327/Debug_T-Beam_SX1262.txt)
[Test_TBeam.ino.txt](https://github.com/jgromes/RadioLib/files/14550337/Test_TBeam.ino.txt)

**Comment by HeadBoffin** [comment]
The examples are in the process of being updated and there has been limited testing on ABP. I'll have to trace the code to see what the positive error message means as it's derived from the join state, which ABP doesn't have.

Can you try the OTAA example please.

And pllease turn off Debug Verbose, it's hard to see the info mixed in with the SPI commands.

**Reply by icicmoi** [reply]
Thanks @HeadBoffin for your return :-)
I'm going to have to admit that I'm too stubborn to insist on this abp mode ;)
Since you asked, I'll create a new otaa node and give the result here ;)

**Reply by StevenCellist** [reply]
This is just a result of outdated examples included in 6.4.2.
2745 decimal in hexadecimal reads 0x0AB9 which is the result code of an active AbP session (note the resemblance of AB9 and ABP).
This can be resolved by changing the line that checks status == ERR_NONE to >= (inequality). This should be done for the line right after beginABP().
If unsure, you can always get the most recent example from here on GitHub, as we're awaiting pending changes for the upcoming 6.4.3 release.

**Reply by icicmoi** [reply]
>  note the resemblance of AB9 and ABP

!  I'm going to need a hexa => qwerty converter !!!!

Thanks again!
With spring just a few days away, I can send a message to the gateway: I'm really happy ... but it would be too easy if I didn't have a few more questions: I'm getting "mismatch, expected 1ec52bcb, got 1d00540e failed, code -7".
/*The calculated and expected CRCs of received packet do not match.
  This means that the packet was damaged during transmission and should be sent again.*/
This is the case for all frames, but my gateway is just a few meters from my module. 
(I borrowed our associative gateway which was too far away !!!)

I've just installed a version in OTAA mode and the problem is exactly the same (frame received and decoded but error 7).

ps : node config class A, Lorawan 1.0.3

---

I'm attaching the .ino file and the debug (otaa & abp)
[Test_Tbeam_OTAA.ino.txt](https://github.com/jgromes/RadioLib/files/14551788/Test_Tbeam_OTAA.ino.txt)
[Debug_Tbeam_OTAA.txt](https://github.com/jgromes/RadioLib/files/14551798/Debug_Tbeam_OTAA.txt)
[Debug_Tbeam_ABP.txt](https://github.com/jgromes/RadioLib/files/14551799/Debug_Tbeam_ABP.txt)

**Reply by HeadBoffin** [reply]
> ps : node config class A, Lorawan 1.0.3

The library is mostly only compatible with v1.1, RP ver B and there are some crucial differences that would make 1.0.3 rather confused.

I'd estimate you are about a week ahead of having it all laid out for you. However you can use the work in progress seen here: https://github.com/HeadBoffin/RadioLib_DevCopy/tree/master/examples/LoRaWAN/Starter which includes extensive notes on setup on TTN.

Even then there is the note on the README about it being in beta. It should be a solid beta once we've ironed out a couple of persistence issues and got the examples online with some notes on the wiki, but right now YMMV.

**Reply by HeadBoffin** [reply]
> but my gateway is just a few meters from my module.

A standard issue when starting out - LoRa radios are really sensitive so if a device & gateway are too close, their input circuitry can get overloaded. So best to have the gateway 5 to 10m away with an intervening brick wall.

**Reply by icicmoi** [reply]
Thank you very much for all this additional information and for the link to your notes: I'll be sure to share it all in our fablab :-)
